src/streamsync/core_ui.py

```diff
@@ -100,6 +103,8 @@ def ingest(self, serialised_components: Dict[str, Any]) -> None:
             self.components.pop(component_id)
         for component_id, sc in serialised_components.items():
             component = Component(**sc)
+            if component.type == "page" and component_id not in self.components:
```

Comment:
Did you think to reuse attach here ?

Reply:
This was what I've tried initially, but it leads to constant re-attachment of the whole tree, including root components. Paints console red, to put it shortly 🥲